enumerables.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength,
+module Enumerable
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_each
+    i = 0
+    return to_enum unless block_given?
+
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    return to_enum unless block_given?
+
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    if block_given?
+      arr = []
+      my_each do |i|
+        arr << i if yield(i)
+      end
+    end
+    arr
+  end
+
+  def my_all?(input = nil)
+    my_each do |i|
+      return false if block_given? && !yield(i)
+
+      if !block_given? && input.nil?
+        return false unless i
+      elsif input
+        return false unless check_input(i, input)
+      end
+    end
+    true
+  end
+
+  def my_any?(input = nil)
+    my_each do |i|
+      return true if block_given? && yield(i)
+
+      if !block_given? && input.nil?
+        return true if i
+      elsif !block_given? && input
+        return true if check_input(i, input)
+      end
+    end
+    false
+  end
+
+  def my_none?(input = nil)
+    my_each do |i|
+      return false if block_given? && yield(i)
+
+      if !block_given? && input.nil?
+        return false if i
+      elsif !block_given? && input
+        return false if check_input(i, input)
+      end
+    end
+    true
+  end
+
+  def my_count(input = nil)
+    counter = 0
+    my_each do |i|
+      if block_given? && input.nil?
+        counter += 1 if yield(i)
+      elsif i && input.nil?
+        counter += 1
+      elsif check_input(i, input) && input.is_a?(Integer)
+        counter += 1
+      end
+    end
+    counter
+  end
+
+  def my_map
+    arr = []
+    my_each do |i|
+      return to_enum unless block_given?
+
+      arr << yield(i) || arr << proc.call(i) if block_given?
+    end
+    arr
+  end
+
+  def my_inject(start = nil, symbol = nil)
+    new_array = to_a
+    memo = start
+    if start && symbol
+      my_each do |i|
```

Comment:
still, there seems to be a small error, when starting a range from a number that is not 1, the results differ compared against the original:

```
irb#1(main):032:0> (5..7).inject(4) { |prod, n| prod * n }
=> 840
irb#1(main):033:0> (5..7).my_inject(4) { |prod, n| prod * n }
=> 168
irb#1(main):034:0> (2..7).my_inject(4) { |prod, n| prod * n }
=> 10080
irb#1(main):035:0> (2..7).inject(4) { |prod, n| prod * n }
=> 20160
irb#1(main):036:0> (1..7).inject(4) { |prod, n| prod * n }
=> 20160
irb#1(main):037:0> (1..7).my_inject(4) { |prod, n| prod * n }
=> 20160

```

Reply:
Thanks Samuel! I've been refactoring that method like crazy to make it work. I just checked what was wrong. Thanks for your reviews!!